File: FrontEnd/src/dto/Position.ts
import type { PositionRiskLevel } from "@/enum/positionRiskLevel";

export class Position {
    id: number | null = null;
    name: string = "";
    risk_level: PositionRiskLevel | null = null
    min_wage: number | null = null;
    max_wage: number | null = null;
    state: boolean = true
    risk_level_text: string = ''
    min_wage_text: string = ''
    max_wage_text: string = ''
}

File: BackEnd/db/models/position.py
from sqlalchemy import Column, Integer, Boolean, String
from sqlalchemy.orm import relationship

from db.database import Base

class Position(Base):
    __tablename__ = "Positions"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String)
    risk_level = Column(Integer)
    min_wage = Column(Integer)
    max_wage = Column(Integer)
    state = Column(Boolean)

    workers = relationship("Worker", back_populates="position")